osf/management/commands/populate_internet_archives_collections.py

```diff
@@ -0,0 +1,90 @@
+import json
+import logging
+
+from osf.models import RegistrationProvider
+from django.core.management.base import BaseCommand
+from website import settings
+import requests
+
+logger = logging.getLogger(__file__)
+
+
+def create_ia_subcollection(provider, version_id, dry_run):
+    provider_id = f'osf-registration-providers-{provider._id}-{version_id}'
+    resp = None
+    if not dry_run:
+        resp = requests.put(
+            f'https://s3.us.archive.org/{provider_id}',
+            headers={
+                'Authorization': f'LOW {settings.IA_ACCESS_KEY}:{settings.IA_SECRET_KEY}',
+                'x-archive-meta01-title': provider.name,
+                'x-archive-meta02-collection': settings.IA_ROOT_COLLECTION,
+                'x-archive-meta03-mediatype': 'collection',
+            },
+        )
+    logger.info(
+        f'{"DRY_RUN" if dry_run else ""} collection for {provider_id} requested with {resp}'
+    )
+    return resp
+
+def modify_ia_subcollection(provider, version_id, dry_run):
+    provider_id = f'osf-registration-providers-{provider._id}-{version_id}'
+    if not dry_run:
+        # The following uses the json patch syntax more info here:
+        # https://archive.org/services/docs/api/metadata.html
+        changes = [
+            {
+                'target': 'metadata',
+                'patch': {
+                    'op': 'replace',
+                    'path': '/title',
+                    'value': provider.name,
+                },
+            },
+        ]
+        resp = requests.post(
+            f'http://archive.org/metadata/{provider_id}',
+            headers={
+                'Authorization': f'LOW {settings.IA_ACCESS_KEY}:{settings.IA_SECRET_KEY}',
+            },
+            data={
+                'target': 'metadata',
+                '-changes': json.dumps(changes),
+                'priority': None,
+            },
+        )
+    logger.info(
```

Comment:
can we have some more angry logs if the call has an error response?

Reply:
Error reporting here makes more sense to me if it just goes to Sentry. Log statements are going to be reductive if it hits a 400 or something.